compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -1257,16 +1258,18 @@ class ResolverImpl(
                 argument.projectionKind != org.jetbrains.kotlin.types.Variance.INVARIANT
             ) {
                 // conflicting variances
-                // TODO: error message
-                return null
+                throw IllegalArgumentException(
```

Comment:
I took liberty in making this an exception. My argument is that returning an error type in a situation seemingly close to an internal error is a bit weird, maybe it's better crash. If we are not ready to make such decision here, please tell me and I'll revert to returning `null` and later on an error type.

Reply:
Conflicting variance might come from incorrectly defined classes and types applications which should be error types to make more sense as it is user input error rather than the compiler internal error. I think your error message below is good though, so maybe adding it to the hint since the hint is for easier error understanding anyways?